### PR TITLE
[SYCL] Reenable enqueue-arg-order tests on CUDA

### PR DESCRIPTION
**Pull request by sergey-semenov**
The tests were disabled due to
https://github.com/intel/llvm/issues/4171, which has been resolved.

### CODE REVIEW

**Review by v-klochkov — APPROVED** [review]

**Comment by sergey-semenov** [comment]
@vladimirlaz I believe @smaslov-intel is the only code owner for these tests, but he's currently on vacation. Can this PR be merged without his approval?

**Pull request merged**
